addons/mail/static/src/models/discuss_public_root/discuss_public_root.js

```diff
@@ -0,0 +1,49 @@
+/** @odoo-module **/
+
+import { attr, one2one } from '@mail/model/model_field';
+import { registerNewModel } from '@mail/model/model_core';
+
+function factory(dependencies) {
+
+    class DiscussPublicRoot extends dependencies['mail.model'] {
+
+        /**
+         * Displays the thread view linked to the discuss public root instead of
+         * the welcome view.
+         */
+        showThreadView() {
+            this.update({ activeView: 'thread' });
+        }
+
+    }
+
+    DiscussPublicRoot.fields = {
+        /**
+         * Determines the view (either 'thread' or 'welcome') to be displayed.
+         */
+        activeView: attr({
+            required: true,
+        }),
```

Comment:
No need for this extra field, just dynamically create and delete the threadView and/or the welcomeView as appropriate and directly make conditions on them.